http-server/src/server.rs

```diff
@@ -124,6 +147,10 @@ pub struct Server {
 	max_request_body_size: u32,
 	/// Access control
 	access_control: AccessControl,
+	/// Pair of channels to stop the server.
+	stop_pair: (mpsc::Sender<()>, mpsc::Receiver<()>),
+	/// Stop handle that indicates whether server has been stopped.
+	stop_handle: Arc<Mutex<()>>,
```

Comment:
Is this mutex really necessary, can't we just call `Sender::is_connected` instead?

As I understand it, the server keeps the `receiver` and when it's closed it's dropped? Thus, then `Sender::is_connected` will fail? Or am I missing something?

Reply:
`is_connected` is not async, unfortunately. Thus it'd be a burden on the user's shoulders to wait for it, and most likely will result in timeout-based polling, which is not the nicest thing in the world.
From my experience, using mutexes as semaphores is kinda normal practice, is there something wrong with it?

Reply:
Alright, it makes sense to have an async API for this.

> From my experience, using mutexes as semaphores is kinda normal practice, is there something wrong with it?

It's fine it's just that `Sender` is kind of already a "semaphore" (it's a built on top of AtomicUsize)